engine/src/main/java/io/camunda/zeebe/engine/state/deployment/DbFormState.java

```diff
@@ -66,11 +71,18 @@ public void storeFormRecord(final FormRecord record) {
     formByIdAndVersionColumnFamily.upsert(idAndVersionKey, dbPersistedForm);
 
     updateLatestVersion(record);
+    updateLatestFormCache();
   }
 
   @Override
   public Optional<PersistedForm> findLatestFormById(
       final DirectBuffer formId, final String tenantId) {
+    final PersistedForm cachedForm =
+        formByTenantAndIdCache.getOrDefault(tenantId, new Object2ObjectHashMap<>()).get(formId);
+    if (cachedForm != null) {
+      return Optional.of(cachedForm);
+    }
```

Comment:
❌ If form record not found in the cache, we should still add it to the cache once we retrieve it from DB.